src/main/java/Duke.java

```diff
@@ -1,10 +1,177 @@
+import java.io.IOException;
+import java.util.Scanner;
+
+/**
+ * Duke is the main class that launches the chat bot when it is run.
+ */
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+
+    /**
+     * This is the path of the file where the tasks are saved to and loaded from.
+     */
+    private String filePath;
+
+    /**
+     * This is the storage class which does the saving and loading of the files.
+     */
+    private Storage storage;
+
+    /**
+     * This is the task list class that contains the list of tasks.
+     */
+    private TaskList tasks;
+
+    /**
+     * The ui class handles the output to the user.
+     */
+    private Ui ui;
+
+    /**
+     * Creates a new chat bot with a specified file path.
+     * If the specified file contains pre-saved tasks, it will be loaded.
+     * @param filePath The path of the file that is used to save and load the tasks.
+     */
+    public Duke(String filePath) {
+        this.filePath = filePath;
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (Exception e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    public void run() throws IOException {
+        this.ui.showLogo();
+        Scanner sc = new Scanner(System.in);
+        boolean chat = true;
+
+        while (chat) {
```

Comment:
chat is a boolean and should sound like one. So maybe isChatting would be a better boolean name? :)

Reply:
You're right! Thanks for the catch!